src/renate/updaters/experimental/er.py

```diff
@@ -80,21 +80,26 @@ def _create_metrics_collections(
 
     def on_model_update_start(
         self, train_dataset: Dataset, val_dataset: Dataset, task_id: Optional[str] = None
-    ) -> Tuple[DataLoader, DataLoader]:
+    ) -> None:
         """Called before a model update starts."""
+        super().on_model_update_start(train_dataset, val_dataset, task_id)
         self._set_memory_loader()
-        self._current_train_dataset = train_dataset
-        train_loader, val_loader = super().on_model_update_start(
-            train_dataset, val_dataset, task_id
+
+    def train_dataloader(self) -> DataLoader:
```

Comment:
Maybe I am missing something, but why cannot you use the one in the Learner base class?

Reply:
We add the enumeration here, which is used for the memory updates after each step.